src/container/streamAnalyticsManager.ts

```diff
@@ -4,40 +4,129 @@
 import StreamAnalyticsManagementClient = require("azure-arm-streamanalytics");
 import { StreamingJob } from "azure-arm-streamanalytics/lib/models";
 import { StreamingJobs } from "azure-arm-streamanalytics/lib/operations";
+import * as fse from "fs-extra";
 import * as request from "request-promise";
 import * as vscode from "vscode";
 import { Constants } from "../common/constants";
 import { UserCancelledError } from "../common/UserCancelledError";
 import { Utility } from "../common/utility";
-import { AzureAccount } from "../typings/azure-account.api";
+import { AzureAccount, AzureSession, AzureSubscription } from "../typings/azure-account.api";
 import { StreamAnalyticsPickItem } from "./models/streamAnalyticsPickItem";
 
+export enum ASAUpdateStatus {
+    Idle,
+    CheckingUpdate,
+    Updating,
+}
+
 export class StreamAnalyticsManager {
+    public static getInstance(): StreamAnalyticsManager {
+        if (!StreamAnalyticsManager.instance) {
+            StreamAnalyticsManager.instance = new StreamAnalyticsManager();
+            return StreamAnalyticsManager.instance;
+        }
+        return StreamAnalyticsManager.instance;
+    }
+
+    private static instance: StreamAnalyticsManager;
     private readonly azureAccount: AzureAccount;
     private readonly MaximumRetryCount: number = 3;
+    private asaUpdateStatus: ASAUpdateStatus;
 
-    constructor() {
+    private constructor() {
         this.azureAccount = vscode.extensions.getExtension<AzureAccount>("ms-vscode.azure-account")!.exports;
+        this.asaUpdateStatus = ASAUpdateStatus.Idle;
     }
 
-    public async selectStreamingJob(): Promise<StreamAnalyticsPickItem> {
+    public async getJobInfo(): Promise<any> {
         await Utility.waitForAzLogin(this.azureAccount);
-
-        const job: StreamAnalyticsPickItem = await vscode.window.showQuickPick(this.loadAllStreamingJobs(), { placeHolder: `Select ${Constants.asaJobDesc}`, ignoreFocusOut: true });
-        if (!job) {
+        const streamingJob: StreamAnalyticsPickItem = await vscode.window.showQuickPick(this.loadAllStreamingJobs(), { placeHolder: `Select ${Constants.asaJobDesc}`, ignoreFocusOut: true });
+        if (!streamingJob) {
             throw new UserCancelledError();
         }
-        return job;
+
+        return await vscode.window.withProgress({
```

Comment:
this part of code is duplicate with the updateJobInfo.

Reply:
They handle two different things, one is read information from user selected ASA Job, and other is read information from template file, I don't think they need to merge together